### PR TITLE
Add unconditional unreachable expression

### PR DESCRIPTION
**Pull request by dschuff**
Add an expression which typechecks in any context, and always traps when
executed.

### CODE REVIEW

File: ml-proto/spec/check.ml [hunk 1]
```diff
@@ -218,6 +218,9 @@ let rec check_expr c et e =
     check_expr c (Some t) e2;
     check_expr c (Some t) e3
 
+  | Unreachable ->
+    check_type None None e.at
```
Comment (dschuff) [inline ml-proto/spec/check.ml]: Halp, I'm an ocaml n00b!
Is there a better way to do nothing here?

Comment (rossberg-chromium) [inline ml-proto/spec/check.ml]: You can simply replace this line with `()`.

**Comment by rossberg-chromium** [comment]
LGTM modulo the one comment.

File: ml-proto/spec/check.ml (hunk 1 above)
Comment (dschuff) [inline ml-proto/spec/check.ml]: Done, thanks,

**Pull request merged**